Runtime/CSharp/Bitcraft.StateMachine.UnitTests/HandWritten/States/TransitionState.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Bitcraft.StateMachine.UnitTests.HandWritten.States
+{
+    public class TransitionState : StateBase
+    {
+        public TransitionState()
+            : base(HandWrittenStateTokens.TransitionStateToken)
+        {
+        }
+
+        protected override void OnInitialized()
+        {
+            base.OnInitialized();
+
+            RegisterActionHandler(HandWrittenActionTokens.TerminateAction, (d, cb) =>
+            {
+                var testContext = (Context as StateMachineTestContext);

```

Comment:
That's why there is the `GetContext<>()` method. We prefer a clean cast exception here than a null ref on next statement.

```suggestion
                var testContext = GetContext<StateMachineTestContext>();
```